File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.py
class Solution:
    def minimumDifference(self, nums: List[int], k: int) -> int:
        if len(nums)==1 and k==1:
            return 0
        n=len(nums)
        maxi=float('inf')
        nums.sort()
        i=0
        j=i+k-1
        while j<n:
            maxi=min(maxi,nums[j]-nums[i])
            i+=1
            j+=1
        return maxi

File: 1984-minimum-difference-between-highest-and-lowest-of-k-scores/1984-minimum-difference-between-highest-and-lowest-of-k-scores.cs
public class Solution {
    public int MinimumDifference(int[] nums, int k) {
        int left=0;
        int right=k-1;
        Array.Sort(nums);
        int res=int.MaxValue;
        while(right<nums.Length){
            res=Math.Min(res,Math.Abs(nums[left]-nums[right]));
            left++;
            right++;
        }
        return res;
        
    }
}